File: sephora-backend/Domain/Entities/ProductPiece.cs
namespace CleanArchitecture.Domain.Entities;

public class ProductPiece
{
    public long Id { get; set; }

    public int? InStock { get; set; }

    public decimal Price { get; set; }

    public int? AmountId { get; set; }

    public Amount? Amount { get; set; }

    public bool IsBottledParfume { get; set; }

    public long ProductId { get; set; }

    public ProductEntity Product { get; set; }
}

File: sephora-frontend/src/models/product/ProductPiece.ts
import ProductPicture from "./ProductPicture.ts";
import Product from "./Product.ts";

export default interface ProductPiece {
    get id(): number;
    get inStock(): number;
    get price(): number;
    get milliliters(): number;
    get isBottledParfume(): boolean;
    get productId(): number;
    get product(): Product;
    get pictures(): ProductPicture[];

    get createdAt(): Date;
    get isNew(): boolean;
}
